decipher/views.py

```diff
@@ -1,6 +1,178 @@
+    @return: Данные, необходимые для скачивания файла
+    """
+    file = file_url
+    file_name = os.path.basename(file)
+    chunk_size = 8192
+    response = StreamingHttpResponse(FileWrapper(open(file, 'rb'), chunk_size),
+                                     content_type=mimetypes.guess_type(file)[0])
+    response['Content-Length'] = os.path.getsize(file)
+    response['Content-Disposition'] = "attachment; filename=%s" % file_name
+    return response
+
+
+def file_save(decipher_data: bytes, file_url):
+    """
+    Осуществляет скачивание файла на сервер
+    @param decipher_data: Расшифрованные данные
+    @param file_url: Путь к файлу на сервере
+    @return: Данные, необходимые для скачивания файла
+    """
+    file_url = file_url[0:-10]
+    with open(file_url, 'wb') as newFile:
+        newFile.write(decipher_data)
+    return download_file(file_url)
+
+
+def xor_file_decryption(cipher_data: bytes, password, file_url):
+    """
+    Осуществляет расшифровку файла с помощью алгоритма XOR
+    @param cipher_data: Зашифрованные данные
+    @param password: Пароль, вводимый пользователем
+    @param file_url: Путь к файлу на сервере
+    @return: Данные, необходимые для скачивания файла
+    """
+    decrypted_data = []
+    for i in range(len(cipher_data)):
+        decrypted_data.append(cipher_data[i] ^ ord(password[i % len(password)]))
+    decipher_data = bytes(decrypted_data)
+    return file_save(decipher_data, file_url)
+
+
+def aes_file_decryption(cipher_data: bytes, password: str, file_url: str):
+    """
+    Осущетсвляет шифрование файла с помощью алгоритма AES
+    @param cipher_data: Зашифрованные данные
+    @param password: Пароль, вводимый пользователем
+    @param file_url: Путь к файлу на сервере
+    @return: Данные, необходимые для скачивания файла
+    """
+    try:
+        key = hashlib.sha256(password.encode('utf-8')).digest()
+        iv = cipher_data[:16]
+        encrypted_data = cipher_data[16:]
+        decipher = AES.new(key, AES.MODE_CBC, iv)
+        decipher_data = unpad(decipher.decrypt(encrypted_data), AES.block_size)
+        return file_save(decipher_data, file_url)
+    except (ValueError, KeyError) as e:
+        print(f"Incorrect decryption. Error: {e}")
+
+
+def file_processing(request, button_value: str):
+    """
+    Осуществляет получение данных из файла
+    @param request: Запрос
+    @param button_value: Значение нажатой кнопки
+    @return: Данные, необходимые для скачивания файла
+    """
+    upload = request.FILES.get('upload', False)
+    fss = FileSystemStorage()
+    file = fss.save(upload.name, upload)
+    file_url = fss.url(file)[1:]
+    with open(file_url, 'rb') as file:
+        data = file.read()
+    password = request.POST.get('filePassword')
+    if button_value == 'AESdcrptFile':
+        return aes_file_decryption(data, password, file_url)
+    elif button_value == 'XORdcrptFile':
+        return xor_file_decryption(data, password, file_url)
```

Comment:
Условие написано правильно, только возвращать внутри него не стоит, лучше сделать это так:
```suggestion
    if button_value == 'AESdcrptFile':
        aes_file_decryption(data, password, file_url)
    elif button_value == 'XORdcrptFile':
        xor_file_decryption(data, password, file_url)
    return create_response(file_url)
```
Функция `create_response` соответствует твоей функции `dowlonad_file`, просто название более подходящее, так как данная функция оборачивает твой зашифрованный файл HTTP ответ, который отправляется на машину клиента.
Тоже самое нужно сделать и для xor